tests/quantization/hqq/test_hqq.py

```diff
@@ -0,0 +1,172 @@
+# coding=utf-8
+# Copyright 2024 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+import unittest
+
+from transformers import AutoModelForCausalLM, AutoTokenizer, HqqConfig
+from transformers.testing_utils import (
+    require_accelerate,
+    require_torch_gpu,
+    slow,
+)
+from transformers.utils import is_accelerate_available, is_hqq_available, is_torch_available
+
+
+if is_torch_available():
+    import torch
+
+if is_accelerate_available():
+    pass
+
+if is_hqq_available():
+    from hqq.core.quantize import HQQBackend, HQQLinear
+
+
+@require_torch_gpu
+class HqqConfigTest(unittest.TestCase):
+    def test_to_dict(self):
+        """
+        Makes sure the config format is properly set
+        """
+        quantization_config = HqqConfig()
+        hqq_orig_config = quantization_config.to_dict()
+
+        for key in hqq_orig_config:
+            self.assertEqual(quantization_config.quant_config[key], hqq_orig_config[key])
+
+
+class HQQLLMRunner:
+    def __init__(self, model_id, quant_config=None, compute_dtype=torch.float16, device="cuda", cache_dir=None):
+        self.model = AutoModelForCausalLM.from_pretrained(
+            model_id,
+            torch_dtype=compute_dtype,
+            device_map=device,
+            quantization_config=quant_config,
+            cache_dir=cache_dir,
+        )
+        self.tokenizer = AutoTokenizer.from_pretrained(model_id, cache_dir=cache_dir)
+        self.device = self.model.device
+        HQQLinear.set_backend(HQQBackend.PYTORCH)
+
+
+def cleanup():
+    torch.cuda.empty_cache()
+    gc.collect()
+
+
+@slow
+@require_torch_gpu
+@require_accelerate
+class HQQTest(unittest.TestCase):
```

Comment:
could you add a multi-gpu test ?

Reply:
Yes, good idea!

Reply:
https://github.com/huggingface/transformers/pull/29637/commits/7a1bbca2ac992313cb64415e4d205f3d1552dc0c
Also refactored the code to make it more re-usable for other tests. Tests pass on my machine